controller/controller.go

```diff
@@ -99,39 +100,111 @@ func (c *Controller) Attach(attachRequest map[string]string) resources.FlexVolum
 	_, err := c.Client.GetVolume(getVolumeRequest)
 
 	if err != nil {
-		return resources.FlexVolumeResponse{
+		return k8sresources.FlexVolumeResponse{
 			Status:  "Failure",
 			Message: "Failed checking volume, call create before attach",
 			Device:  volumeName}
 
 	}
 
-	return resources.FlexVolumeResponse{
+	return k8sresources.FlexVolumeResponse{
 		Status:  "Success",
 		Message: "Volume already attached",
 		Device:  volumeName,
 	}
 
 }
 
+//GetVolumeName checks if volume is attached
+func (c *Controller) GetVolumeName(getVolumeNameRequest k8sresources.FlexVolumeGetVolumeNameRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-isAttached-start")
+	defer c.logger.Println("controller-isAttached-end")
+	volumeName, ok := getVolumeNameRequest.Opts["volumeName"]
+	if !ok {
+		return k8sresources.FlexVolumeResponse{
+			Status:  "Failure",
+			Message: fmt.Sprintf("Failed checking volumeName, ", getVolumeNameRequest.Opts),
+		}
+
+	}
+	return k8sresources.FlexVolumeResponse{
+		Status:     "Success",
+		Message:    "Volume is attached",
+		VolumeName: volumeName,
+		Device:     volumeName,
+		Attached:   true,
+	}
+}
+
+//WaitForAttach Waits for a volume to get attached to the node
+func (c *Controller) WaitForAttach(waitForAttachRequest k8sresources.FlexVolumeWaitForAttachRequest) k8sresources.FlexVolumeResponse {
```

Comment:
@shay-berman , I am still experimenting with k8s 1.6/1.7, I will respond to your questions once I have at least a first running version to understand what is optional and what is required